src/artemis/system_tests/test_fgs_plan.py

```diff
@@ -0,0 +1,142 @@
+from unittest.mock import MagicMock, patch
+
+import bluesky.preprocessors as bpp
+import pytest
+from bluesky.run_engine import RunEngine
+
+from artemis.devices.eiger import EigerDetector
+from artemis.devices.fast_grid_scan_composite import FGSComposite
+from artemis.external_interaction.callbacks import FGSCallbackCollection
+from artemis.fast_grid_scan_plan import get_plan, read_hardware_for_ispyb, run_gridscan
+from artemis.parameters import SIM_BEAMLINE, DetectorParams, FullParameters
+
+
+@pytest.fixture()
+def eiger() -> EigerDetector:
+    detector_params: DetectorParams = DetectorParams(
+        current_energy=100,
+        exposure_time=0.1,
+        directory="/tmp",
+        prefix="file_name",
+        detector_distance=100.0,
+        omega_start=0.0,
+        omega_increment=0.1,
+        num_images=50,
+        use_roi_mode=False,
+        run_number=0,
+        det_dist_to_beam_converter_path="src/artemis/devices/unit_tests/test_lookup_table.txt",
+    )
+    eiger = EigerDetector(
+        detector_params=detector_params, name="eiger", prefix="BL03S-EA-EIGER-01:"
+    )
+
+    # Otherwise odin moves too fast to be tested
+    eiger.cam.manual_trigger.put("Yes")
+
+    # S03 currently does not have StaleParameters_RBV
+    eiger.wait_for_stale_parameters = lambda: None
+    eiger.odin.check_odin_initialised = lambda: (True, "")
+
+    yield eiger
+
+
+params = FullParameters()
+params.beamline = SIM_BEAMLINE
+
+
+@pytest.fixture
+def RE():
+    return RunEngine({})
+
+
+@pytest.fixture
+def fgs_composite():
+    fast_grid_scan_composite = FGSComposite(
+        insertion_prefix=params.insertion_prefix,
+        name="fgs",
+        prefix=params.beamline,
+    )
+    return fast_grid_scan_composite
+
+
+@pytest.mark.s03
+@patch("artemis.fast_grid_scan_plan.wait_for_fgs_valid")
+@patch("bluesky.plan_stubs.wait")
+@patch("bluesky.plan_stubs.kickoff")
+@patch("bluesky.plan_stubs.complete")
+def test_run_gridscan(
```

Comment:
Nit: This test still doesn't work for me but I think it's outside the scope of this issue so going to put a skip on it and approve this PR.